src/main/java/com/temaula/rdb/Periodo.java

```diff
@@ -0,0 +1,64 @@
+package com.temaula.rdb;
+
+import com.temaula.rdb.constraints.CustomConstraint;
+
+import javax.persistence.Embeddable;
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+import javax.validation.constraints.NotNull;
+import java.io.Serializable;
+import java.time.LocalDate;
+import java.util.Objects;
+import java.util.Optional;
+
+@Embeddable
+@CustomConstraint(
+        delegateTo = Periodo.PeriodoValidator.class
+)
+public class Periodo implements Serializable {
+
+    @NotNull
+    public LocalDate dataInicio;
+    @NotNull
+    public LocalDate dataFim;
+
+    public static Periodo of(LocalDate dataInicio, LocalDate dataFim) {
```

Comment:
Acho que ele só deveriar criar uma instancia de periodo se ele for válido.
Do contrário, uma exceção.

Reply:
Esse método é utilizado só pelos testes por enquanto... nem precisaria dele agora no momento pois a instanciação é realizada pelo parsing do body da request pelo framework e o Bean Validation garante a validação...